src/vm/jitinterface.cpp

```diff
@@ -4591,8 +4591,77 @@ TypeCompareState CEEInfo::compareTypesForCast(
         CORINFO_CLASS_HANDLE        fromClass,
         CORINFO_CLASS_HANDLE        toClass)
 {
-    // Stub for now
-    return TypeCompareState::May;
+    CONTRACTL {
+        SO_TOLERANT;
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    TypeCompareState result = TypeCompareState::May;
+
+    JIT_TO_EE_TRANSITION();
+
+    TypeHandle fromHnd = (TypeHandle) fromClass;
```

Comment:
You may want to rename:

fromClass =>  fromHnd 
toClass => toHnd

And

fromHnd => fromType
toHnd => toType

Reply:
As handle implies that we are treating the local as an opaque thing.

Reply:
Runtime code doesn't seem super-consistent here; I'll see if Jan has a preference.